src/sdk/namespace_fs.js

```diff
@@ -2766,9 +2766,11 @@ class NamespaceFS {
                 }
                 return version_info;
             } catch (err) {
-                retries -= 1;
-                if (retries <= 0 || !native_fs_utils.should_retry_link_unlink(is_gpfs, err)) throw err;
                 dbg.warn(`NamespaceFS._delete_single_object_versioned: retrying retries=${retries} file_path=${file_path}`, err);
+                retries -= 1;
+                // retry also on ENOENT if the version moved from .versions -> latest or the opposite/deleted concurrently
```

Comment:
can you maybe elaborate on what exactly is the issue that might cause this. its not clear from the comment

Reply:
it's written in the description but I updated the comment as well, please take a look